app/controllers/users_controller.rb

```diff
@@ -0,0 +1,62 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: [:show]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user= User.find_by(id: params[:id])
+    if @user.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    username = params[:user][:username]
+    @user = User.find_by(username: username)
+
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Successfully logged in as returning user #{username}"
+    else
+      @user = User.create(username: username, joined_date: Time.now)
+      
+      if @user.save
+        session[:user_id] = @user.id
+        flash[:success] = "Successfully logged in as new user #{username}"
+      else
+        render :login_form
+        return
+      end
+    end
+
+    redirect_to root_path
+  end 
+
+  def current
+    @current_user = User.find_by(id: session[:user_id])
+    unless @current_user
+      flash[:error] = "You must be logged in to see this page"
+      redirect_to root_path
+    end
+  end 
+
+  def logout
+    flash[:success] = "Successfully logged out."
+    session[:user_id] = nil
+    redirect_to root_path
+  end
+
+
+  private
+
+  def find_user
```

Comment:
Consider putting this method in your application controller so other controllers will also inherit this method.